src/Application/Asset/TerminalAssetPersistenceService.php

```
@@ -0,0 +1,116 @@
+            $this->cacheTerminalStatus($asset),
+        );
+    }
+
+    /**
+     * @template TResult of object
+     *
+     * @param \Closure(string, AssetStatus): TResult $staleAssetResultFactory
+     *
+     * @return TResult|null
+     */
+    private function saveAsset(Asset $asset, \Closure $staleAssetResultFactory): ?object
```

PHPMD (2.15.0)

[warning] Avoid excessively long variable names like $staleAssetResultFactory. Keep variable name length under 20. (undefined)

(LongVariable)